bolt-modules/boltlib/lib/puppet/datatypes/target.rb

```diff
@@ -20,7 +20,8 @@
       vars => { type => Optional[Hash[String[1], Data]], kind => given_or_derived },
       facts => { type => Optional[Hash[String[1], Data]], kind => given_or_derived },
       features => { type => Optional[Array[String[1]]], kind => given_or_derived },
-      plugin_hooks => { type => Optional[Hash[String[1], Data]], kind => given_or_derived }
+      plugin_hooks => { type => Optional[Hash[String[1], Data]], kind => given_or_derived },
```

Comment:
It seems like most of these will never be nil, are they optional just in case?

Reply:
I'm actually not sure -- the `Target` datatype is defined a little differently from the rest, so I just followed what was there. @donoghuc might have a better idea since it looks like he originally wrote it as optional?